src/main/java/macrobase/analysis/transform/aggregate/BatchedWindowAggregate.java

```diff
@@ -0,0 +1,12 @@
+package macrobase.analysis.transform.aggregate;
+
+import macrobase.datamodel.Datum;
+import java.util.List;
+
+public abstract class BatchedWindowAggregate implements IncrementalWindowAggregate {
```

Comment:
What if instead this class implements `process(List<Datum> new_data, List<Datum> expired_data)` (from `IncrementalWindowAggregate`?

Basic idea: BatchedWindowAggregate can keep track of the current window, then, each time `process(new_data, expired_data)` is called, it updates the window and passes the entire window to an abstract `public Datum process(List<Datum> data)` function.